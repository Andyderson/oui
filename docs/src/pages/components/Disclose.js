
  import React from 'react';
  import ComponentContent from '../../layouts/templates/ComponentContent';

   import examples from '../../../../src/components/Disclose/example'; import readme from 'raw-loader!../../../../src/components/Disclose/README.md';

  const sassData = {"name":"Disclose\n","description":"Similar to an accordion but any number of items can be opened\nclosed independently.","start":{"line":1,"column":1},"end":{"line":18,"column":4},"example":[{"type":"html","description":null,"code":"<div class=\"#{$namespace}disclose is-active\">\n  <a href=\"#\" class=\"#{$namespace}disclose__link link--dark\">\n    <div class=\"#{$namespace}disclose__arrow is-active\">\n      <span class=\"#{$namespace}disclose__symbol\"></span>\n      Title of Disclosure\n    </div>\n  </a>\n  <div class=\"#{$namespace}disclose__content\">\n    <h3>Some Title</h3>\n    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis accusantium corporis, beatae maxime quasi. Tempora.</p>\n  </div>\n</div>"}]};
  const reactData = {"description":"","displayName":"Disclose","methods":[{"name":"handleToggle","docblock":null,"modifiers":[],"params":[{"name":"event","type":null}],"returns":null}],"props":{"children":{"type":{"name":"node"},"required":true,"description":""},"childrenStyle":{"type":{"name":"string"},"required":false,"description":""},"headerStyle":{"type":{"name":"string"},"required":false,"description":""},"noBorder":{"type":{"name":"bool"},"required":false,"description":""},"title":{"type":{"name":"string"},"required":true,"description":""}},"private":false};

  let dataObject = {
    react: reactData, 
    examples: examples ? examples : null,
    readme: readme ? readme : null,
    sass: sassData,
  }
  
  class DiscloseComponent extends React.Component {
  
    render() {
      return (
        <ComponentContent data={ dataObject } />
      );
    };
  }
  
  export default DiscloseComponent
