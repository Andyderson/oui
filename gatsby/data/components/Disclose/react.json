{"description":"","displayName":"Disclose","methods":[{"name":"handleToggle","docblock":null,"modifiers":[],"params":[{"name":"event","type":null}],"returns":null}],"props":{"children":{"type":{"name":"node"},"required":true,"description":""},"childrenStyle":{"type":{"name":"string"},"required":false,"description":""},"headerStyle":{"type":{"name":"string"},"required":false,"description":""},"noBorder":{"type":{"name":"bool"},"required":false,"description":""},"title":{"type":{"name":"string"},"required":true,"description":""}},"private":false}