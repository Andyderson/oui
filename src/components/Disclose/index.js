import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

class Disclose extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      overChildren: false,
    };
    this.handleToggle = this.handleToggle.bind(this);
  }

  handleToggle(event) {
    this.setState({
      isOpen: !this.state.isOpen,
      overChildren: false,
    });
  }

  // boxed
  // lego-disclose__link link--dark soft-half border--all background--faint display--block flush

  // lego-disclose__content soft border--sides border--bottom

  // border--bottom

  render() {
    const { children, style } = this.props;
    const contentClass = classNames(
      'lego-disclose__content',
      {
        ['soft border--sides border--bottom']: style === 'all',
        ['border--bottom']: style === 'divider',
      }
    );
    const linkClass = classNames(
      'lego-disclose__link link--dark soft-half flush',
      {
        ['border--all background--faint display--block']: style === 'all' || style === 'header',
      }
    );
    const arrow = this.state.isOpen ? 'lego-disclose is-active' : 'lego-disclose';
    
    return (
      <div className={ arrow } style={{marginTop: '-1px'}}>
        <a onClick={ this.handleToggle } className={ linkClass }>
          <div className='lego-disclose__arrow'>
            <span className="lego-disclose__symbol"></span>
            Title of Disclosure
          </div>
        </a>
        <div className={ contentClass }>
          { this.state.isOpen && children }
        </div>
      </div>
    );
  }
}

Disclose.propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.string,
};

export default Disclose;
