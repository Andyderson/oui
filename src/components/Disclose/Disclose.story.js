import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import Disclose from './index.js';

const stories = storiesOf('Disclose', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.add('default', withInfo()(() => {
  return (
    <Disclose title='default title'>
      <h3>Some Title</h3>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
      expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
      accusantium corporis, beatae maxime quasi. Tempora.</p>
    </Disclose>
  );
}));

stories.add('multiple stacked', withInfo()(() => {
  return (
    <div>
      <Disclose headerStyle='header-bordered' title='Some Title' childrenStyle='border'>
        <div className='soft--left'>
          <h3>Some Title</h3>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
          expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
          accusantium corporis, beatae maxime quasi. Tempora.</p>
        </div>
      </Disclose>
      <Disclose headerStyle='header-bordered' title='Some Title'>
        <div className='soft--left'>
          <h3>Some Title</h3>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
          expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
          accusantium corporis, beatae maxime quasi. Tempora.</p>
        </div>
      </Disclose>
    </div>
  );
}));

stories.add('header style', withInfo()(() => {
  return (
    <div>
      <Disclose headerStyle='header' title='Browsers'>
        <div className='soft--left'><a>Safari</a></div>
        <div className='soft--left'><a>Chrome</a></div>
        <div className='soft--left'><a>Opera</a></div>
        <div className='soft--left'><a>Please stop using IE</a></div>
      </Disclose>
      <Disclose headerStyle='header' title='Some Title' childrenStyle='border'>
        <div className='soft--left'>
          <h3>Some Title</h3>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
          expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
          accusantium corporis, beatae maxime quasi. Tempora.</p>
        </div>
      </Disclose>
    </div>
  );
}));

stories.add('with divider', withInfo()(() => {
  return (
    <div>
      <Disclose childrenStyle='divider' title='Some Title'>
        <h3>Some Title</h3>
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
        expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
        accusantium corporis, beatae maxime quasi. Tempora.</p>
      </Disclose>
      <Disclose childrenStyle='divider' title='Some Title'>
        <h3>Some Title</h3>
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
        expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
        accusantium corporis, beatae maxime quasi. Tempora.</p>
      </Disclose>
    </div>
  );
}));

