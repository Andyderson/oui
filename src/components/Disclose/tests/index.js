import React from 'react';
import { shallow } from 'enzyme';
import Disclose from '../index';

const disclose = shallow(<Disclose>un disclose</Disclose>);

describe('Disclose Component', () => {
  it('renders correctly', () => {
    expect(disclose).toMatchSnapshot();
  });

  it('initialize with `state` isOpen is false', () => {
    expect(disclose.state().isOpen).toBe(false);
  });

  it('will set `state` isOpen to true when click on link', () => {
    disclose.find('.lego-disclose__link').simulate('click');
    expect(disclose.state().isOpen).toBe(true);
  });
});
