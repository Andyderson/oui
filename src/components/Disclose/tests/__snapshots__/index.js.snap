// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclose Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Disclose>
    un disclose
  </Disclose>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="lego-disclose__link link--dark soft-half flush"
          onClick={[Function]}
        >
          <div
            className="lego-disclose__arrow"
          >
            <span
              className="lego-disclose__symbol"
            />
            Title of Disclosure
          </div>
        </a>,
        <div
          className="lego-disclose__content"
        />,
      ],
      "className": "lego-disclose",
      "style": Object {
        "marginTop": "-1px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="lego-disclose__arrow"
          >
            <span
              className="lego-disclose__symbol"
            />
            Title of Disclosure
          </div>,
          "className": "lego-disclose__link link--dark soft-half flush",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="lego-disclose__symbol"
              />,
              "Title of Disclosure",
            ],
            "className": "lego-disclose__arrow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "className": "lego-disclose__symbol",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "Title of Disclosure",
          ],
          "type": "div",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "lego-disclose__content",
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="lego-disclose__link link--dark soft-half flush"
            onClick={[Function]}
          >
            <div
              className="lego-disclose__arrow"
            >
              <span
                className="lego-disclose__symbol"
              />
              Title of Disclosure
            </div>
          </a>,
          <div
            className="lego-disclose__content"
          />,
        ],
        "className": "lego-disclose",
        "style": Object {
          "marginTop": "-1px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="lego-disclose__arrow"
            >
              <span
                className="lego-disclose__symbol"
              />
              Title of Disclosure
            </div>,
            "className": "lego-disclose__link link--dark soft-half flush",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="lego-disclose__symbol"
                />,
                "Title of Disclosure",
              ],
              "className": "lego-disclose__arrow",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "lego-disclose__symbol",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "Title of Disclosure",
            ],
            "type": "div",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "lego-disclose__content",
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
